src/Cake.Core/IO/NuGet/IPackageReferenceBundler.cs

```diff
@@ -0,0 +1,17 @@
+using System.Collections.Generic;
+
+namespace Cake.Core.IO.NuGet
+{
+    /// <summary>
+    /// Bundles packageAssemblyFiles into package reference sets grouped by their target framework.
+    /// </summary>
+    public interface IPackageReferenceBundler
+    {
+        /// <summary>
+        /// Bundles the provided packageAssemblyFiles into package reference sets grouped by their target framework.
+        /// </summary>
+        /// <param name="packageAssemblyFiles">The package assembly files.</param>
+        /// <returns>The newly created PackageReferenceSets</returns>
+        IEnumerable<PackageReferenceSet> BundleByTargetFramework(IEnumerable<FilePath> packageAssemblyFiles);
```

Comment:
Maybe this should return a materialized list/collection such as `IReadOnlyList`/`IReadOnlyCollection`?